### PR TITLE
feat(gitea): Add PR cache

### PR DESCRIPTION
**Pull request by zharinov**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]

**Comment by viceice** [comment]
import order issues

**Comment by viceice** [comment]
conflicted 🙃

**Comment by viceice** [comment]
@zharinov Can you continue on this? I really like to cache the PR data to reduce Gitea / Forgejo load.

**Comment by zharinov** [comment]
> @zharinov Can you continue on this? I really like to cache the PR data to reduce Gitea / Forgejo load.

Yes, but `gitea-helper.ts` and the way it's mocked makes it hard to test, so I have to perform huge test refactoring first

**Comment by viceice** [comment]
ok, let me know if I can help anywhere

**Comment by zharinov** [comment]
- Blocked by #26451

**Comment by zharinov** [comment]
@viceice Don't you know the reason why `memCache` is impossible to initialize for testing? The object is always `undefined` even if `init()` was called in `beforeEach`. It seems to work for `cache/memory.spec.ts` itself, but not for modules that use it transitively.

**Comment by viceice** [comment]
> @viceice Don't you know the reason why `memCache` is impossible to initialize for testing? The internal object holding key-values is always `undefined` even if `init()` was called in `beforeEach`. It seems to work for `cache/memory.spec.ts` itself, but not for modules that use it transitively.

🤔 maybe some global mocking?

**Comment by zharinov** [comment]
- Blocked by #26634

File: lib/modules/platform/gitea/index.spec.ts [hunk 1]
```diff
@@ -192,6 +193,7 @@ describe('modules/platform/gitea/index', () => {
   beforeEach(async () => {
     jest.resetModules();
```
Comment (viceice) [inline lib/modules/platform/gitea/index.spec.ts:194]: this was causing your issue, so you need to use dynamic import to get the right memcache 🙃

File: lib/util/cache/memory/index.ts [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
+  return global.renovateMemCache?.[key];
 }
 
 export function set(key: string, value: unknown): void {
-  if (repoCache) {
-    repoCache[key] = value;
+  if (global.renovateMemCache) {
+    global.renovateMemCache[key] = value;
   }
 }
```
Comment (viceice) [inline lib/util/cache/memory/index.ts:23]: please revert this

File: lib/modules/platform/gitea/pr-cache.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+      url = parseLinkHeader(res.headers.link)?.next?.url;
+    }
+
+    return this;
+  }
+
+  private toRenovatePR(data: PR): Pr | null {
+    return toRenovatePR(data, this.author);
+  }
```
Comment (viceice) [inline lib/modules/platform/gitea/pr-cache.ts:143]: can we inline this? I see only one usage

**Review by viceice — COMMENTED** [review]
just one small thing

**Review by viceice — APPROVED** [review]

**Pull request merged**